client/src/unifyfs-sysio.c

```diff
@@ -616,27 +638,51 @@ int UNIFYFS_WRAP(remove)(const char* path)
         /* get file id for path name */
         int fid = unifyfs_get_fid_from_path(upath);
         if (fid < 0) {
-            /* ERROR: file does not exist */
-            LOGDBG("Couldn't find entry for %s in UNIFYFS", upath);
-            errno = ENOENT;
-            return -1;
-        }
+            /* file doesn't exist locally, but may exist globally */
+            int gfid = unifyfs_generate_gfid(upath);
+            unifyfs_file_attr_t attr = {0};
 
-        /* check that it's not a directory */
-        if (unifyfs_fid_is_dir(fid)) {
-            /* TODO: shall be equivalent to rmdir(path) */
-            /* ERROR: is a directory */
-            LOGDBG("Attempting to remove a directory %s in UNIFYFS", upath);
-            errno = EISDIR;
-            return -1;
-        }
+            int ret = unifyfs_get_global_file_meta(gfid, &attr);
+            if (ret != UNIFYFS_SUCCESS) {
+                /* ERROR: file does not exist locally or globally */
+                LOGDBG("Couldn't find entry for %s in UNIFYFS", upath);
+                errno = ENOENT;
+                return -1;
+            }
 
-        /* shall be equivalent to unlink(path) */
-        /* delete the file */
-        int ret = unifyfs_fid_unlink(fid);
-        if (ret != UNIFYFS_SUCCESS) {
-            errno = unifyfs_rc_errno(ret);
-            return -1;
+            /* check that it's not a directory */
+            if (attr.mode & S_IFDIR) {
+                /* TODO: shall be equivalent to rmdir(path) */
+                /* ERROR: is a directory */
+                LOGDBG("Attempting to unlink a directory %s in UNIFYFS", upath);
+                errno = EISDIR;
+                return -1;
+            }
```

Comment:
I think it's valid to call ``remove`` on a directory.  I see we also had this check for a directory in our ``remove`` wrapper before this change.
```
NAME
       remove - remove a file or directory

SYNOPSIS
       #include <stdio.h>

       int remove(const char *pathname);

DESCRIPTION
       remove() deletes a name from the file system.  It calls unlink(2) for files, and rmdir(2) for directories.
```

Reply:
Updated this for when the file is found locally with the following. However, haven't added anything to address this when the dir is found globally and this brought up a few questions. I'll explain this in my comment below.
``` C
         } else { /* got the file locally */
            /* if directory, is it empty? */
            if (unifyfs_fid_is_dir(fid) && !unifyfs_fid_is_dir_empty(upath)) {
                /* ERROR: is a directory, but isn't empty */
                LOGDBG("Attempting to remove non-empty dir %s in UNIFYFS",
                        upath);
                errno = ENOTEMPTY;
                 return -1;
             }
```